### PR TITLE
Build examples executable with dune

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR ~renames the created example executable~, builds the example with dune, and updates the documentation on how to use that.

The idea is to make it even easier to see this thing running.

I hope this helps! 🌴

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,10 +54,11 @@ This depends on cohttp and can serve the metrics collected above over HTTP.
 
 The `prometheus-app.unix` ocamlfind library provides the `Prometheus_unix` module,
 which includes a cmdliner option and pre-configured web-server.
-See the `examples/example.ml` program for an example, which can be run as:
+See the `examples/mirage_prometheus_example.ml` program for an example, which can be run as:
 
 ```shell
-$ ./_build/examples/example.native --listen-prometheus=9090
+$ dune build
+$ dune exec -- examples/mirage_prometheus_example.exe --listen-prometheus=9090
```
Comment (talex5) [inline README.md]: You don't need the `dune build` here - `exec` does that for you.

File: examples/dune [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(executable
+  (name         mirage_prometheus_example)
+  (package      prometheus)
+  (public_name  mirage-prometheus-example)
```
Comment (talex5) [inline examples/dune]: We probably don't want `public_name` here. There's not much point in installing this binary to the user's path.

But even if we do, there's no need to rename the source file too - `name` doesn't need to match `public_name`.

**Review by talex5 — COMMENTED** [review]
Thanks - looks like this got forgotten in the move to dune!

Travis is complaining, though:

```
# File "examples/dune", line 5, characters 27-46:
# 5 |   (libraries    prometheus prometheus-app.unix cmdliner cohttp-lwt cohttp-lwt-unix))
#                                ^^^^^^^^^^^^^^^^^^^
# Error: Library "prometheus-app.unix" not found.
```

File: examples/dune (hunk 2 above)
Comment (olleolleolle) [reply examples/dune]: What you describe is what I attempted (and failed) to do. I've probably misunderstood some part of this.

Comment (olleolleolle) [reply examples/dune]:
```
File "examples/dune", line 3, characters 2-27:
3 |   (package      prometheus)
      ^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This field is useless without a (public_name ...) field.
```

(This, I believe, is what led me to add `public_name`.

Removing `(package      prometheus)` 

```
File "examples/dune", line 2, characters 16-41:
2 |   (name         mirage_prometheus_example)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Module "Mirage_prometheus_example" doesn't exist.
```

image

Could you tell me where to step from there?

Comment (olleolleolle) [reply examples/dune]: I'll punt to having the name `examples/example.ml`.

Comment (olleolleolle) [reply examples/dune]: Force-pushed a more minimal change, avoiding the install step.

**Comment by talex5** [comment]
Looks good - thanks!

**Comment by olleolleolle** [comment]
@talex5 Thank _you_, I learned things!

**Pull request merged**